sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -217,12 +218,17 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
   }
 
   override fun addWidget(widget: Widget) {
+    val weakThis = WeakReference(this)
+    widget.setRepaintTrigger {
+      weakThis.get()?.renderer?.scheduleRepaint()
+    }
     renderer.renderThread.addWidget(widget)
     renderer.scheduleRepaint()
   }
 
   override fun removeWidget(widget: Widget): Boolean {
     val wasRemoved = renderer.renderThread.removeWidget(widget)
+    widget.setRepaintTrigger(null)
```

Comment:
Should this be inside the `if (wasRemoved)`?

Reply:
fixed